src/world_diff.rs

```diff
@@ -102,6 +108,7 @@ impl WorldDiff {
             .copied()
             .unwrap_or_else(|| world.read_storage(contract, key).unwrap_or_default());
 
+        self.read_storage_slots_ct.add((contract, key));
```

Comment:
Instead of adding this new field, you can just change the code so that free storage slots are inserted into `read_storage_slots`, too.